open-x-htb-system-tests.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
```

Comment:
I've followed your guidelines here -https://knowledgebase.indexexchange.com/display/ADAPTER/Test+Case+Functions , but __generateRequestObj(returnParcels) in our adapter is never called within your test suite. I also used the example provided in the doc to check, but got failing test results for test case like 'should construct bid request correctly' because the request firing never got triggered. Could you provide some advice on this?

Reply:
The above issues could be related to the `getConfig` function in `open-x-htb-system-tests.js`.
More information can be found <here> https://knowledgebase.indexexchange.com/display/ADAPTER/Test+Case+Functions#TestCaseFunctions-BidRequestTestCase
The ad slot configurations need to be nested under the `xSlots` property (see example in link above).
You need to define only two items in `xSlots`, not three. 
The required partner-level property `host` is missing.
 
Once updated, it should look as follows:
```javascript
return {
    host: '<valid host>', // The required partner-level property.
    xSlots: { // All the ad slot-level properties under the `xSlots` property.
        1: {
            adUnitId: '54321',
            sizes: [[300, 250], [300, 600]]
        },
        2: {
            adUnitId: '12345',
            sizes: [[300, 600]]
        }
    }
};
```
 
This will allow the tests to run and for you to breakpoint in the adapter code. Failures that occur after the aforementioned changes are made will most likely be related to those tests cases that were written under the assumption that three items existed under `xSlots`. For example, in `validateBidRequest` it expects there to be three `auid`’s, however, there will only be two.